src/v/datalake/batching_parquet_writer.h

```diff
@@ -75,4 +76,20 @@ class batching_parquet_writer : public data_writer {
     data_writer_result _result;
 };
 
+class batching_parquet_writer_factory : public data_writer_factory {
```

Comment:
optional: could have also made this a nested class so it would be `batching_parquet_writer::factory`